### PR TITLE
Removed unused code from `spec/support`

### PR DESCRIPTION
**Pull request by almorbah**
Resolves [APPEALS-41034](https://jira.devops.va.gov/browse/APPEALS-41034)

# Description
Removed unused code from `spec/support`

## Acceptance Criteria
- [ ] Code compiles correctly

## Testing Plan

1. Go to [Jira Issue/Test Plan Link](https://jira.devops.va.gov/browse/JIRA-12345) or list them below

- [ ] For feature branches merging into master: Was this deployed to UAT?

# Frontend
## User Facing Changes
 - [ ] Screenshots of UI changes added to PR & Original Issue

 BEFORE|AFTER
 ---|---

## Storybook Story
*For Frontend (Presentation) Components*
* [ ] Add a [Storybook](https://github.com/department-of-veterans-affairs/caseflow/wiki/Documenting-React-Components-with-Storybook) file alongside the component file (e.g. create `MyComponent.stories.js` alongside `MyComponent.jsx`)
* [ ] Give it a title that reflects the component's location within the overall Caseflow hierarchy
* [ ] Write a separate story (within the same file) for each discrete variation of the component

# Backend
## Database Changes
*Only for Schema Changes*

* [ ] Add typical timestamps (`created_at`, `updated_at`) for new tables
* [ ] Update column comments; include a "PII" prefix to indicate definite or potential [PII data content](https://github.com/department-of-veterans-affairs/appeals-team/blob/master/caseflow-team/0-how-we-work/pii-handbook.md#what-is-pii)
* [ ] Have your migration classes inherit from `Caseflow::Migration`, especially when adding indexes (use `add_safe_index`) (see [Writing DB migrations](https://github.com/department-of-veterans-affairs/caseflow/wiki/Writing-DB-migrations))
* [ ] Verify that `migrate:rollback` works as desired ([`change` supported functions](https://edgeguides.rubyonrails.org/active_record_migrations.html#using-the-change-method))
* [ ] Perform query profiling (eyeball Rails log, check bullet and fasterer output)
* [ ] For queries using raw sql was an explain plan run by System Team
* [ ] Add appropriate indexes (especially for foreign keys, polymorphic columns, unique constraints, and Rails scopes)
* [ ] Run `make check-fks`; add any missing foreign keys or add to `config/initializers/immigrant.rb` (see [Record associations and Foreign Keys](https://github.com/department-of-veterans-affairs/caseflow/wiki/Record-associations-and-Foreign-Keys))
* [ ] Add `belongs_to` for associations to enable the [schema diagrams](https://department-of-veterans-affairs.github.io/caseflow/task_trees/schema/schema_diagrams) to be automatically updated
* [ ] Document any non-obvious semantics or logic useful for interpreting database data at [Caseflow Data Model and Dictionary](https://github.com/department-of-veterans-affairs/caseflow/wiki/Caseflow-Data-Model-and-Dictionary)

## Integrations: Adding endpoints for external APIs
* [ ] Check that Caseflow's external API code for the endpoint matches the code in the relevant integration repo
  * [ ] Request: Service name, method name, input field names
  * [ ] Response: Check expected data structure
  * [ ] Check that calls are wrapped in MetricService record block
* [ ] Check that all configuration is coming from ENV variables
  * [ ] Listed all new ENV variables in description
  * [ ] Worked with or notified System Team that new ENV variables need to be set
* [ ] Update Fakes
* [ ] For feature branches: Was this tested in Caseflow UAT

# Best practices
## Code Documentation Updates
- [ ] Add or update code comments at the top of the class, module, and/or component.

### Test Coverage
Did you include any test coverage for your code? Check below:
- [ ] RSpec
- [ ] Jest
- [ ] Other

### Code Climate
Your code does not add any new code climate offenses? If so why?
- [ ] No new code climate issues added

### Monitoring
- [ ] Are performance metrics (e.g., response time, throughput) being tracked?
- [ ] Are key application components monitored (e.g., database, cache, queues)?
- [ ] Is there a system in place for setting up alerts based on performance thresholds?

### Logging
- [ ] Are logs being produced at appropriate log levels (debug, info, warn, error, fatal)?
- [ ] Are logs structured (e.g., using log tags) for easier querying and analysis?
- [ ] Are sensitive data (e.g., passwords, tokens) redacted or omitted from logs?
- [ ] Is log retention and rotation configured correctly?
- [ ] Are logs being forwarded to a centralized logging system if needed?

### Auditing
- [ ] Are user actions being logged for audit purposes?
- [ ] Are changes to critical data being tracked ?
- [ ] Are logs being securely stored and protected from tampering or exposing protected data?

### Error Handling
- [ ] Are errors being caught and handled gracefully?
- [ ] Are appropriate error messages being displayed to users?
- [ ] Are critical errors being reported to an error tracking system (e.g., Sentry, ELK)?
- [ ] Are unhandled exceptions being caught at the application level ?

### Exception Handling
- [ ] Are custom exceptions defined and used where appropriate?
- [ ] Is exception handling consistent throughout the codebase?
- [ ] Are exceptions logged with relevant context and stack trace information?
- [ ] Are exceptions being grouped and categorized for easier analysis and resolution?

### CODE REVIEW

File: spec/support/download_helper.rb [hunk 1]
```diff
@@ -17,11 +17,6 @@ def download
     downloads.first
   end
 
-  def download_content
```
Comment (TylerBroyles) [inline spec/support/download_helper.rb:20]: This is potentially useful even if it isn't currently being used in any feature tests.

File: spec/support/feature_helper.rb [hunk 2]
```diff
@@ -60,10 +60,6 @@ def step(title)
     yield
   end
 
-  def xstep(title)
```
Comment (TylerBroyles) [inline spec/support/feature_helper.rb:63]: Both of these methods are debugging tools. They wouldn't show up in the code even if people use them daily. They probably shouldn't be removed.

File: spec/support/feature_helper.rb [hunk 3]
```diff
@@ -93,11 +93,6 @@ def be_titled(title)
     have_xpath("//title[contains(.,'#{title}')]", visible: false)
   end
 
-  def hang
```
Comment (pamatyatake2) [inline spec/support/feature_helper.rb:96]: I feel like this method might be useful for debugging propose. I feel like keeping this method won't hurt.

File: spec/support/intake_helpers.rb [hunk 4]
```diff
@@ -265,10 +260,6 @@ def select_intake_no_match
     safe_click ".add-issue"
   end
 
-  def get_claim_id(claim_review)
```
Comment (pamatyatake2) [inline spec/support/intake_helpers.rb:268]: I am seeing few use cases where this method could be used in 
```
spec/models/ramp_refiling_intake_spec.rb:348
spec/models/ramp_refiling_intake_spec.rb:349
spec/models/ramp_refiling_intake_spec.rb:479

``` 
but i will let you decide if you want to remove this method or use it in some of these cases.

Comment (almorbah) [reply spec/support/intake_helpers.rb:268]: That looks like a reasonable place to use this helper method. I found two other places which I used it for.

File: spec/support/feature_helper.rb (hunk 3 above)
Comment (pamatyatake2) [reply spec/support/feature_helper.rb:96]: Yes, debugger would be used if you are not running the rspec test and just debugging in the console. this method i think was created to debug the frontend issues when running the rspec. lets assume your test is failing for some reason and then you need to  debug in the browser developer console then you can use this method to hang your rspec process and inspect the frontend code in browser.

File: spec/support/stubbable_user.rb [hunk 5]
```diff
@@ -28,16 +28,6 @@ def authenticate!(css_id: nil, roles: nil, user: nil)
       self.stub = user
     end
 
-    def tester!(roles: nil)
```
Comment (TylerBroyles) [inline spec/support/stubbable_user.rb:31]: This method is used in another test file.

Comment (almorbah) [reply spec/support/stubbable_user.rb:31]: I couldn't see that file in our code base.

**Review by pamatyatake2 — APPROVED** [review]
LGTM

**Review by brandondorner — APPROVED** [review]

**Pull request merged**